ci/linux_openresty_1_19_runner.sh

```diff
@@ -17,5 +17,5 @@
 #
 
 
-export OPENRESTY_VERSION=1.19.3.2
+export OPENRESTY_VERSION=1.19.9.1
```

Comment:
We have promised to support the latest three versions of OpenResty, so we still need to cover 1.19.3.2.
What about updating the test case and only running it when the version is matched, like https://github.com/apache/apisix/blob/release/2.14/t/plugin/ext-plugin/sanity-openresty-1-19.t?

It would be better to refuse to start when the OpenResty is too old and k8s discovery is used in the stream.

Reply:
> We have promised to support the latest three versions of OpenResty, so we still need to cover 1.19.3.2.
What about updating the test case and only running it when the version is matched, like https://github.com/apache/apisix/blob/release/2.14/t/plugin/ext-plugin/sanity-openresty-1-19.t?

ok, i will try it later.

> It would be better to refuse to start when the OpenResty is too old and k8s discovery is used in the stream.

Is it a good choice if we add a boolean field like `enable_stream`(default is `false`) into `apisix/discovery/kubernetes/schema.lua` as a property and check OpenResty version if `enable_stream` is `true`? Can you give me some hints? thanks.

Reply:
something like https://github.com/apache/apisix/blob/84535aff0e5bca575202df45833197958b93fe40/apisix/upstream.lua#L425

Reply:
@spacewander 

- first, it will cause err like below when start apisix with openresty 1.19.3.2, because `local discovery = require("apisix.discovery.init").discovery` in `apisix/upstream.lua:19` will import `discovery/kubernetes/init.lua`.

```
$ ./bin/apisix start
/usr/local/openresty-debug//luajit/bin/luajit ./apisix/cli/apisix.lua start
nginx: [error] init_by_lua error: /usr/local/openresty-debug/lualib/ngx/process.lua:5: unsupported subsystem: stream
stack traceback:
        [C]: in function 'error'
        /usr/local/openresty-debug/lualib/resty/core/base.lua:223: in function 'allows_subsystem'
        /usr/local/openresty-debug/lualib/ngx/process.lua:5: in main chunk
        [C]: in function 'require'
        ...ng/Documents/apisix/apisix/discovery/kubernetes/init.lua:29: in main chunk
        [C]: in function 'require'
        /home/ronething/Documents/apisix/apisix/discovery/init.lua:28: in main chunk
        [C]: in function 'require'
        /home/ronething/Documents/apisix/apisix/upstream.lua:19: in main chunk
        [C]: in function 'require'
        /home/ronething/Documents/apisix/apisix/http/service.lua:18: in main chunk
        [C]: in function 'require'
        /home/ronething/Documents/apisix/apisix/init.lua:35: in main chunk
        [C]: in function 'require'
        init_by_lua:3: in main chunk
```

- then, something like `if conf.pass_host == "node" and conf.nodes and ` in `check_upstream_conf` function may only print error log when check upstream if `in_dp` is true, or cause err when post/put upstream or other resource which need to check upstream_conf both in http and stream subsystem. But we need to refuse to start apisix when the OpenResty is too old and k8s discovery is used in the stream?

so maybe we should consider other solutions? if something i was wrong, please tell me, thanks.

Reply:
> But we need to refuse to start apisix when the OpenResty is too old and k8s discovery is used in the stream?

Yes. Unlike the upstream conf which is a runtime conf, the wrong discovery conf in the yaml can't be corrected without restarting APISIX.

And we need to provide a clear error message so people don't need to debug it.